### DISCUSSION TITLE
Update the Alexa Device state and value from ESP

### DISCUSSION THREAD

**Discussion by RCr0ssley** [vintlabs/fauxmoESP#251] [Q&A]
Hi,

I've got a project controlling using the fauxmo library to control a fan (Would be nice to have the device show as a fan not lamp but it works). I have external controls that also switch on / off the fan as well as using Alexa commands. Is there a way to update the device on Alexa so that it shows if the device has been changed locally, i.e switched off via the button.

I thought I'd seen an example of this before but can't seem to find it now.

Many thanks

**Comment by AmaralSza** [comment]
Good afternoon, I managed to figure out how to change the "type" of equipment?

**Comment by ddweber456** [comment]
@RCr0ssley , if your button in connected to the ESP device and you can know its state of that button (ON/OFF  TRUE/FALSE), then you can programatically set the state and value and use the fauxmo. setState call as shown on line 148 of the first example code (fauxmoESP_Basic.ino)
